### PR TITLE
ddl: disallow dropping auto_increment column attribute

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve? 

Disallow dropping auto_increment column attribute

### What is changed and how it works?

Tweak the check in `getModifiableColumnJob`.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

Side effects

Related changes

Release note

### CODE REVIEW

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -417,6 +417,9 @@ type SessionVars struct {
 
 	// PrevStmt is used to store the previous executed statement in the current session.
 	PrevStmt string
+
+	// AllowDropAutoInc indicates whether a user can drop the auto_increment column attribute or not.
+	AllowDropAutoInc bool
```
Comment (bb7133) [inline sessionctx/variable/session.go]:
```suggestion
	AllowRemoveAutoInc bool
```
IMO using `Remove` is better, `Drop` reminds the users that data are dropped(like `DROP TABLE` or `DROP COLUMN`, but this is not the case)

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -2690,6 +2690,10 @@ func (d *ddl) getModifiableColumnJob(ctx sessionctx.Context, ident ast.Ident, or
 	if !mysql.HasAutoIncrementFlag(col.Flag) && mysql.HasAutoIncrementFlag(newCol.Flag) {
 		return nil, errUnsupportedModifyColumn.GenWithStackByArgs("set auto_increment")
 	}
+	// Disallow modifying column from auto_increment to not auto_increment if the session variable `AllowDropAutoInc` is false.
+	if !ctx.GetSessionVars().AllowDropAutoInc && mysql.HasAutoIncrementFlag(col.Flag) && !mysql.HasAutoIncrementFlag(newCol.Flag) {
+		return nil, errUnsupportedModifyColumn.GenWithStackByArgs("drop auto_increment column attribute")
```
Comment (bb7133) [inline ddl/ddl_api.go]: How about adding a tip in the error message that setting `tidb_allow_drop_auto_inc` if the user truly want to remove the attribute

Comment (tangenta) [reply ddl/ddl_api.go]: Good idea.

**Pull request merged**